src/persist/src/indexed/columnar.rs

```diff
@@ -38,25 +39,36 @@ pub const KEY_VAL_DATA_MAX_LEN: usize = i32::MAX as usize;
 
 const BYTES_PER_KEY_VAL_OFFSET: usize = 4;
 
-/// A set of ((Key, Val), Time, i64) records stored in a columnar
+/// A set of ((Key, Val), Time, Diff) records stored in a columnar
 /// representation.
 ///
 /// Note that the data are unsorted, and unconsolidated (so there may be
-/// multiple instances of the same ((Key, Val), Time), and some i64s might be
+/// multiple instances of the same ((Key, Val), Time), and some Diffs might be
 /// zero, or add up to zero).
 ///
+/// Both Time and Diff are presented externally to persist users as a type
+/// parameter that implements [mz_persist_types::Codec64]. Our columnar format
+/// intentionally stores them both as i64 columns (as opposed to something like
+/// a fixed width binary column) because this allows us additional compression
+/// options. Note that we use an i64 for Time (currently a u64 in mz) because
+/// the useful range (nanos since unix epoch) fits in [i64::MAX] until some time
```

Comment:
I thought our time coordinates were in milliseconds since the unix epoch for real time sources? 

One potential complication here is that other source types may present with other time coordinates (like postgres sources using the binlog coordinate for time) which may not always fit neatly in the i64 range? I'm not totally sure how persist will be used by storage I guess and just want to flag that we may not want to assume that time == millis or nanos since epoch every time.

Reply:
I think I just explained it poorly. is this any better?

Reply:
yup thats much better!